internal/pkg/cli/app_show.go

```diff
@@ -0,0 +1,312 @@
+		prodLengthMax = int(math.Max(float64(prodLengthMax), float64(len(strconv.FormatBool(opts.envs[ind].Prod)))))
+		pathLengthMax = int(math.Max(float64(pathLengthMax), float64(len(opts.uris[ind].Path))))
+		urlLengthMax = int(math.Max(float64(urlLengthMax), float64(len(opts.uris[ind].DNSName))))
+	}
+	fmt.Fprintf(writer, "%s\t%s\t%s\t%s\n", strings.Repeat("-", envLengthMax), strings.Repeat("-", prodLengthMax), strings.Repeat("-", pathLengthMax), strings.Repeat("-", urlLengthMax))
+	for ind := range opts.envs {
+		fmt.Fprintf(writer, "%s\t%s\t%s\t%s\n", opts.envs[ind].Name, strconv.FormatBool(opts.envs[ind].Prod), opts.uris[ind].Path, opts.uris[ind].DNSName)
+	}
+	writer.Flush()
+}
+
+func (opts *ShowAppOpts) jsonOutput() (string, error) {
+	type serializedEnv struct {
+		Name   string `json:"name"`
+		Region string `json:"region"`
+		Prod   bool   `json:"prod"`
+		URL    string `json:"url"`
+		Path   string `json:"path"`
+	}
+	// TODO: add service information after we add ECS package to query ECS task information.
+	// type serializedService struct {
+	// 	Memory string `json:"memory"`
+	// 	CPU    string `json:"cpu"`
+	// 	Tasks  string `json:"tasks"`
+	// 	Port   string `json:"port"`
+	// 	ARN    string `json:"arn"`
+	// }
+	type serializedApp struct {
+		AppName      string          `json:"appName"`
+		Type         string          `json:"type"`
+		Project      string          `json:"project"`
+		Account      string          `json:"account"`
+		Environments []serializedEnv `json:"environments"`
+		// Service      serializedService `json:"service"`
+	}
+	environments := make([]serializedEnv, len(opts.envs))
+	for ind := range opts.envs {
+		environments[ind] = serializedEnv{
+			Name:   opts.envs[ind].Name,
+			Region: opts.envs[ind].Region,
+			Prod:   opts.envs[ind].Prod,
+			URL:    opts.uris[ind].DNSName,
+			Path:   opts.uris[ind].Path,
+		}
+	}
+	b, err := json.Marshal(serializedApp{
+		AppName:      opts.app.Name,
+		Type:         opts.app.Type,
+		Project:      opts.ProjectName(),
+		Account:      opts.proj.AccountID,
+		Environments: environments,
+		// Service: serializedService{},
+	})
+	if err != nil {
+		return "", fmt.Errorf("marshal applications: %w", err)
+	}
+	return fmt.Sprintf("%s\n", b), nil
+}
+
+func (opts *ShowAppOpts) selectProject() (string, error) {
+	projNames, err := opts.retrieveProjects()
+	if err != nil {
+		return "", err
+	}
+	if len(projNames) == 0 {
+		log.Infoln("There are no projects to select.")
+		return "", nil
+	}
+	proj, err := opts.prompt.SelectOne(
+		applicationShowProjectNamePrompt,
+		applicationShowProjectNameHelpPrompt,
+		projNames,
+	)
+	if err != nil {
+		return "", fmt.Errorf("selecting projects: %w", err)
+	}
+	return proj, nil
+}
+
+func (opts *ShowAppOpts) selectApplication() (string, error) {
+	appNames, err := opts.retrieveApplications()
+	if err != nil {
+		return "", err
+	}
+	if len(appNames) == 0 {
+		log.Infoln("There are no applications to select.")
+		return "", nil
+	}
```

Comment:
So if there are no applications, what do we display an empty table? or will `Execute` error out?

Maybe it's best to error here if we're going to error down the line